4-HTTP/client/main.js

```diff
@@ -0,0 +1,209 @@
+    if (item.isFinished) {
+      todoDescription.classList.remove(classes.finishedTodo);
+      todoDescription.classList.add(classes.unfinishedTodo);
+      item.isFinished = false;
+    } else {
+      todoDescription.classList.remove(classes.unfinishedTodo);
+      todoDescription.classList.add(classes.finishedTodo);
+      item.isFinished = true;
+    }
+    try {
+      await todosAPI.editTodo(item, item.id);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const createButton = (cb, icon, className) => {
+    const button = document.createElement("button");
+    const iTag = document.createElement("i");
+    iTag.className = icon;
+    button.append(iTag);
+    button.classList.add(className);
+    button.addEventListener("click", cb);
+    button.style.cursor = "pointer";
+    return button;
+  };
+
+  const createSwitchElem = (cb) => {
+    const switchElem = document.createElement("label");
+    switchElem.classList.add("switch");
+
+    const checkbox = document.createElement("input");
+    checkbox.type = "checkbox";
+    checkbox.onclick = cb;
+
+    const slider = document.createElement("span");
+    slider.classList.add("slider");
+    slider.classList.add("round");
+
+    switchElem.appendChild(checkbox);
+    switchElem.appendChild(slider);
+    return switchElem;
+  };
+
+  const createTextItem = (item, switchElem) => {
+    const todoDescription = document.createElement("span");
+    todoDescription.innerText = item.text;
+    todoDescription.classList.add(classes.todoText);
+    if (item.isFinished) {
+      todoDescription.classList.add(classes.finishedTodo);
+      switchElem.firstChild.checked = true;
+    }
+    return todoDescription;
+  };
+
+  const createListItem = (item) => {
+    const listItemContainer = document.createElement("div");
+    listItemContainer.classList.add(classes.listItemText);
+    const switchElem = createSwitchElem(() => toggleTodo(item));
+    listItemContainer.appendChild(switchElem);
+    listItemContainer.appendChild(createTextItem(item, switchElem));
+    return listItemContainer;
+  };
+
+  const createItem = (list, item) => {
+    const todoListItemContainer = document.createElement("li");
+    todoListItemContainer.id = item.id;
+    const listItem = createListItem(item);
+
+    const itemActions = document.createElement("span");
+    itemActions.classList.add(classes.listItemActions);
+    itemActions.appendChild(
+      createButton(() => removeTodo(item), "fa fa-trash", classes.button)
+    );
+    itemActions.appendChild(
+      createButton(() => editTodo(item), "fa fa-pencil", classes.edit)
+    );
+
+    todoListItemContainer.appendChild(listItem);
+    todoListItemContainer.appendChild(itemActions);
+    return todoListItemContainer;
+  };
+
+  const editListenerHandler = async (e) => {
+    const editInputElem = e.target;
+    const id = editInputElem.id.split("-editTodo")[0];
+    try {
+      const { data } = await todosAPI.getTodos();
```

Comment:
I did that because I need the todo item and in this function I got only partial object. I improved this function now so I don't have to fetch all the items.